jsonrpc/endpoints_zkevm.go

```diff
@@ -427,87 +427,107 @@ func (z *ZKEVMEndpoints) GetExitRootsByGER(globalExitRoot common.Hash) (interfac
 		if err != nil {
-			return RPCErrorResponse(types.DefaultErrorCode, "failed to get L2 gas price", err, false)
+			return nil, nil, types.NewRPCError(types.DefaultErrorCode, "failed to encode tx", err, false)
 		}
 
-		txGasPrice := new(big.Int).SetUint64(gasPrices.L2GasPrice) // by default we assume the tx gas price is the current L2 gas price
-		txEGPPct := state.MaxEffectivePercentage
-		egpEnabled := z.pool.EffectiveGasPriceEnabled()
-
-		if egpEnabled {
-			rawTx, err := state.EncodeTransactionWithoutEffectivePercentage(*tx)
-			if err != nil {
-				return RPCErrorResponse(types.DefaultErrorCode, "failed to encode tx", err, false)
-			}
+		txEGP, err := z.pool.CalculateEffectiveGasPrice(rawTx, txGasPrice, gasEstimation, gasPrices.L1GasPrice, gasPrices.L2GasPrice)
+		if err != nil {
+			return nil, nil, types.NewRPCError(types.DefaultErrorCode, "failed to calculate effective gas price", err, false)
+		}
 
-			txEGP, err := z.pool.CalculateEffectiveGasPrice(rawTx, txGasPrice, gasEstimation, gasPrices.L1GasPrice, gasPrices.L2GasPrice)
+		if txEGP.Cmp(txGasPrice) == -1 { // txEGP < txGasPrice
+			// We need to "round" the final effectiveGasPrice to a 256 fraction of the txGasPrice
+			txEGPPct, err = z.pool.CalculateEffectiveGasPricePercentage(txGasPrice, txEGP)
 			if err != nil {
-				return RPCErrorResponse(types.DefaultErrorCode, "failed to calculate effective gas price", err, false)
-			}
-
-			if txEGP.Cmp(txGasPrice) == -1 { // txEGP < txGasPrice
-				// We need to "round" the final effectiveGasPrice to a 256 fraction of the txGasPrice
-				txEGPPct, err = z.pool.CalculateEffectiveGasPricePercentage(txGasPrice, txEGP)
-				if err != nil {
-					return RPCErrorResponse(types.DefaultErrorCode, "failed to calculate effective gas price percentage", err, false)
-				}
-				// txGasPriceFraction = txGasPrice/256
-				txGasPriceFraction := new(big.Int).Div(txGasPrice, new(big.Int).SetUint64(256)) //nolint:gomnd
-				// txGasPrice = txGasPriceFraction*(txEGPPct+1)
-				txGasPrice = new(big.Int).Mul(txGasPriceFraction, new(big.Int).SetUint64(uint64(txEGPPct+1)))
+				return nil, nil, types.NewRPCError(types.DefaultErrorCode, "failed to calculate effective gas price percentage", err, false)
 			}
-
-			log.Infof("[EstimateFee] finalGasPrice: %d, effectiveGasPrice: %d, egpPct: %d, l2GasPrice: %d, len: %d, gas: %d, l1GasPrice: %d",
-				txGasPrice, txEGP, txEGPPct, gasPrices.L2GasPrice, len(rawTx), gasEstimation, gasPrices.L1GasPrice)
+			// txGasPriceFraction = txGasPrice/256
+			txGasPriceFraction := new(big.Int).Div(txGasPrice, new(big.Int).SetUint64(256)) //nolint:gomnd
+			// txGasPrice = txGasPriceFraction*(txEGPPct+1)
+			txGasPrice = new(big.Int).Mul(txGasPriceFraction, new(big.Int).SetUint64(uint64(txEGPPct+1)))
 		}
 
-		fee := new(big.Int).Mul(txGasPrice, new(big.Int).SetUint64(gasEstimation))
+		log.Infof("[EstimateFee] finalGasPrice: %d, effectiveGasPrice: %d, egpPct: %d, l2GasPrice: %d, len: %d, gas: %d, l1GasPrice: %d",
```

Comment:
Maybe is better to use [internalEstimateGasPrice] in the logs of this function instead of [EstimateFee] I used this at it was the name of the function, but now it's changed. Other option is tu use [EstimageGasPrice] remove "internal" word